### ISSUE TITLE
Rendertarget as a resource for a sprite component.

### ISSUE THREAD

**Issue by Dragosha** [defold/defold#8689]
I tried a new feature from version 1.7.0 - create a rendertarget as a resource.  It works in the render script.
But I can't add such a rendertarget to the texture field for a sprite. 

![2024-03-21_15-14-32](https://github.com/defold/defold/assets/842807/81febaa1-2e2d-44ad-b2f9-9bc755011c2d)

My sprites accept two samplers, the second one for render target (shadowmap). The field only accepts atlases and tilesource. Also I can't keep the field empty because I get a build error _`tex1` must be specified._

![2024-03-21_10-20-17](https://github.com/defold/defold/assets/842807/ea416214-9a95-4d74-8970-a639c6e1fac5)

In case of 3d model it works, but there is some issue with display in editor (in runtime everything works fine).

![2024-03-21_15-17-02](https://github.com/defold/defold/assets/842807/9fad6f11-14f5-484d-aa81-e59d6db9bf36)

https://github.com/defold/defold/assets/842807/a596ab8b-789e-4e26-8c5e-7db8efeac6b7

<hr/>

<table>
<tr><td>Build time</td><td>2024-03-12T10:26:40.890206</td></tr>
<tr><td>Defold channel</td><td>editor-alpha</td></tr>
<tr><td>Defold editor sha</td><td>9ad2d7834a6adfaedcd902a8e6f9997ed5cbb507</td></tr>
<tr><td>Defold engine sha</td><td>bf4dc66ab5fbbafd4294d32c2797c08b630c0be5</td></tr>
<tr><td>Defold version</td><td>1.7.0</td></tr>
<tr><td>GPU</td><td>NVIDIA GeForce GTX 1060 6GB/PCIe/SSE2</td></tr>
<tr><td>GPU Driver</td><td>4.6.0 NVIDIA 551.23</td></tr>
<tr><td>Java version</td><td>17.0.5+8</td></tr>
<tr><td>OS arch</td><td>amd64</td></tr>
<tr><td>OS name</td><td>Windows 10</td></tr>
<tr><td>OS version</td><td>10.0</td></tr>
<table>

**Comment by Dragosha** [comment]
as example I used this project: https://github.com/Dragosha/defold-light-and-shadows

**Comment by Jhonnyg** [comment]
Yes this isn’t something new or something broken, render targets act as a texture and not an atlas. We currently don’t support assigning textures into sprites directly, so we need to discuss if that’s something we want or not as it kinda becomes a slippery slope of changes needed in both engine and editor. I.e we will have to start supporting textures for all component types if we do this.

And regarding the preview, we don’t run render scripts in the editor so right now we just create a dummy texture with a solid color for the preview rendering

**Comment by koenbollen** [comment]
This issue is preventing us from having dynamic shadows (in our 2D game). 

Ideally this would work:
1. Render all shadow objects to a separate rendertarget/texture.
2. Render the game, attach this shadow-render-target to the pipeline so the fragment shader of our decals can darken their pixels if they overlap with a shadow.

(this is a very easy and straightforward way)

This now is not possible since I can't attach a (global) extra texture when drawing sprites (like you would in a post-shader using a model).

And I couldn't figure out a way to do this differently (while looking good).

Having the option to `render.enable_texture("shadow_sampler")` for sprites would be the solution.

**Comment by Dragosha** [comment]
> Having the option to `render.enable_texture("shadow_sampler")` for sprites would be the solution.

This works for sprites as well as for other components if the sprite material has a second sampler. The problem is that adding this sampler requires selecting an atlas in the editor, (I just duplicate an first atlas in this field). Although it is overwritten in the renderscript.

**Comment by koenbollen** [comment]
> > Having the option to `render.enable_texture("shadow_sampler")` for sprites would be the solution.
> 
> This works for sprites as well as for other components if the sprite material has a second sampler. The problem is that adding this sampler requires selecting an atlas in the editor, (I just duplicate an first atlas in this field). Although it is overwritten in the renderscript.

I got an test working now. Thanks for the info. 👌
Would be nice if the interface wouldn't require an atlas to be set.
